packages/openmrs/src/Utils.js

```diff
@@ -51,43 +62,84 @@ export async function request(state, method, path, options = {}) {
   };
 
   const url = `${baseUrl}${path}`;
+  let queryParams = opts?.query;
+  const requestOptions = queryParams ? { ...opts, query: queryParams } : opts;
+
+  return (shouldPaginate({ underGlobalLimit: true, hasPageSize: query.pageSize }))
```

Comment:
We should paginate if:
1) it's a GET
2) The number of downloaded items is less than the number of requested items
3) The user is not controlling pagination (ie, the user passed `startIndex`)